### PR TITLE
Init lua plugin with CI and add sub commands: CopilotChatExplain and CopilotChatTests

### PR DESCRIPTION
**Pull request by jellydn**
# What

- Init lua plugin
- Add CI for generate doc
- Add demo usage on readme
- Small improvements: 
	- Go to end of file and open a new line instead of current line
	- Add `CopilotChatExplain` command
	- Add `CopilotChatTests` command

# Demo
- Explain code [Image from Gyazo](https://gyazo.com/4589e35e1935a67b812fa4eade3ef7e4)
- Write unit tests [Image from Gyazo](https://gyazo.com/58396063c97eeecdf85b56e29eed1e3a)

P/S: This is not fully functional like VS Code extension as it doesn't support workspace (refer to below image)

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,12 +10,25 @@ It will prompt you with instructions on your first start. If you already have `C
 
 1. `pip install python-dotenv requests pynvim prompt-toolkit`
 2. Put it in your lazy setup
+
 ```lua
 require('lazy').setup({
-    'gptlang/CopilotChat.nvim',
+  {
+    "jellydn/CopilotChat.nvim",
```
Comment (jellydn) [inline README.md:17]:
```suggestion
    "gptlang/CopilotChat.nvim",
```

File: README.md [hunk 2]
```diff
@@ -24,7 +37,7 @@ require('lazy').setup({
 1. Put the files in the right place
 
 ```
-$ git clone https://github.com/gptlang/CopilotChat.nvim
+$ git clone https://github.com/jellydn/CopilotChat.nvim
```
Comment (jellydn) [inline README.md:40]:
```suggestion
$ git clone https://github.com/gptlang/CopilotChat.nvim
```

**Review by jellydn — COMMENTED** [review]
Hi @gptlang Let me know if you have any thoughts on comment on this PR? Thanks.

**Comment by gptlang** [comment]
I'm currently away from home. I think there are specific prompts used by Copilot Chat on VScode for the special commands. I still prefer `:CopilotChat` over `:CChat`

Once I get back, I plan to port the whole thing into pure lua (remove Python) and add the other features. I'll just leave this open for now.

**Comment by jellydn** [comment]
> I still prefer :CopilotChat over :CChat

I am okay with that. Will revert the command name.

> Once I get back, I plan to port the whole thing into pure lua (remove Python) and add the other features. I'll just leave this open for now as reference.

Cool. Sounds a good plan.

**Pull request closed without merging**